slidechain/watch.go

```diff
@@ -207,7 +197,11 @@ func (c *Custodian) watchPegOuts(ctx context.Context, pegouts <-chan pegOut) {
 			if !ok {
 				log.Fatalf("peg-outs channel closed")
 			}
-			err := c.doPostPegOut(ctx, p.AssetXDR, p.Anchor, p.TxID, p.Amount, p.Seqnum, p.State, p.Exporter, p.TempAddr, p.Pubkey)
+			// This peg-out struct is marshalled into a JSON, which must equal the reference data in the export tx.
+			// That requires that the post-peg-out struct txid be empty.
```

Comment:
There are a couple of better ways to do this. One is not to export the txid field, which causes the json package to skip over it. The other is to give the field the struct tag `json:"-"` which explicitly instructs the json package to skip it.